src/openforms/submissions/models/submission.py

```diff
@@ -469,8 +469,34 @@ def has_registrator(self):
     def is_completed(self):
         return bool(self.completed_on)
 
+    @property
+    def is_ready_to_hash_identifying_attributes(self) -> bool:
+        """
+        Check if the submission can safely hash the identifying auth attributes.
+        """
+        # no authentication -> there's nothing to hash
+        if not self.is_authenticated:
+            return False
+        # completed, but not cosigned yet -> registration after cosigning requires
+        # unhashed attributes
+        if self.is_completed and self.waiting_on_cosign:
+            return False
+
+        # the submission has not been submitted/completed/finalized, so it will
+        # definitely not be processed yet -> okay to hash
+        if not self.is_completed:
```

Comment:
my gut feeling says "pausing", but that *does* hash them deliberately since resuming them allows to convert to plain text again.

This situation happens in a number of cases though:

* user logs out: there's no way to recover the submission, so we must hash
* session times out due to inactivity: no way to recover either -> we can hash

the former will try to hash, the latter won't, but there could potentially be a future celery task taking care of that, processing the submissions that have been "idle" for longer than the session timeout duration.

however, with cosign, the submission _is_ completed and we may not hash yet, because we're waiting for cosign. Completed means "confirmation page was seen and submitted, and the submission can not be altered anymore"